ui/pages/institutional/interactive-replacement-token-page/interactive-replacement-token-page.js

```diff
@@ -0,0 +1,349 @@
+import { useCopyToClipboard } from '../../../hooks/useCopyToClipboard';
+
+const getButtonLinkHref = ({ address }) => {
+  const url = SWAPS_CHAINID_DEFAULT_BLOCK_EXPLORER_URL_MAP[CHAIN_IDS.MAINNET];
+  return `${url}address/${address}`;
+};
+
+export default function InteractiveReplacementTokenPage({ history }) {
+  const dispatch = useDispatch();
+  const isMountedRef = useRef(false);
+  const mmiActions = mmiActionsFactory();
+  const { address } = useSelector((state) => state.metamask.modal.props);
+  const {
+    selectedAddress,
+    custodyAccountDetails,
+    interactiveReplacementToken,
+    mmiConfiguration,
+  } = useSelector((state) => state.metamask);
+  const { custodianName } =
+    custodyAccountDetails[toChecksumHexAddress(address || selectedAddress)] ||
+    {};
+  const { url } = interactiveReplacementToken || {};
+  const { custodians } = mmiConfiguration;
+  const custodian =
+    custodians.find((item) => item.name === custodianName) || {};
+  const mostRecentOverviewPage = useSelector(getMostRecentOverviewPage);
+  const metaMaskAccounts = useSelector(getMetaMaskAccounts);
+  const connectRequests = useSelector(
+    (state) => state.metamask.institutionalFeatures?.connectRequests,
+  );
+  const [isLoading, setIsLoading] = useState(false);
+  const [tokenAccounts, setTokenAccounts] = useState([]);
+  const [error, setError] = useState(false);
+  const t = useI18nContext();
+  const [copied, handleCopy] = useCopyToClipboard();
+  const {
+    removeAddTokenConnectRequest,
+    setCustodianNewRefreshToken,
+    getCustodianAccounts,
+  } = mmiActions;
+  const connectRequest = connectRequests ? connectRequests[0] : undefined;
+
+  useEffect(() => {
+    isMountedRef.current = true;
+    return () => (isMountedRef.current = false);
+  }, []);
+
+  useEffect(() => {
+    const getTokenAccounts = async () => {
+      if (!connectRequest) {
+        history.push(mostRecentOverviewPage);
+        return;
+      }
+
+      try {
+        const custodianAccounts = await dispatch(
+          getCustodianAccounts(
+            connectRequest.token,
+            connectRequest.apiUrl,
+            connectRequest.service,
+            false,
+          ),
+        );
+
+        const filteredAccounts = custodianAccounts.filter(
+          (account) => metaMaskAccounts[account.address.toLowerCase()],
+        );
+
+        const mappedAccounts = filteredAccounts.map((account) => ({
+          address: account.address,
+          name: account.name,
+          labels: account.labels,
+          balance:
+            metaMaskAccounts[account.address.toLowerCase()]?.balance || 0,
+        }));
+
+        if (isMountedRef.current) {
+          setTokenAccounts(mappedAccounts);
+          setIsLoading(false);
+        }
+      } catch (e) {
+        console.log(e);
+        setError(true);
+        setIsLoading(false);
+      }
+    };
+
+    getTokenAccounts();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I would advocate to list all the dependencies and not ignoring them unless it is really needed. Or this is only calculated one time?

Reply:
Yes, this is only calculated one time as we are getting the needed accounts and we don't need to rely on new changes

Reply:
Okay. Makes sense. Can you add some comments in the code to just explain a bit of the reason behind empty dependencies ? Can be beneficial for other devs to understand as well.